client/packages/core/src/instaql.js

```diff
@@ -88,11 +88,23 @@ function refAttrPat(makeVar, store, etype, level, label) {
   return [nextEtype, nextLevel, attrPat, attr, isForward];
 }
 
-function matchesLikePattern(value, pattern) {
-  if (typeof value !== "string" || typeof pattern !== "string") return false;
+function makeLikeMatcher(caseSensitive, pattern) {
+  if (typeof pattern !== "string") {
+    return function likeMatcher(_value) {
+      return false;
+    };
+  }
   const regexPattern = pattern.replace(/%/g, ".*").replace(/_/g, ".");
-  const regex = new RegExp(`^${regexPattern}$`);
-  return regex.test(value);
+  const regex = new RegExp(
+    `^${regexPattern}$`,
+    caseSensitive ? undefined : "i",
+  );
+  return function likeMatcher(value) {
```

Comment:
This should be a little more efficient, because we reuse the regex across matches instead of creating a new one each time.